packages/@aws-cdk/aws-servicecatalog/lib/private/product-stack-synthesizer.ts

```diff
@@ -16,7 +25,17 @@ export class ProductStackSynthesizer extends cdk.StackSynthesizer {
   }
 
   public addFileAsset(_asset: cdk.FileAssetSource): cdk.FileAssetLocation {
-    throw new Error('Service Catalog Product Stacks cannot use Assets');
+    if (!this.stack) {
+      throw new Error('You must call bindStack() first');
+    }
+
+    if (!this.assetBucket) {
+      const parentStack = (this.stack as ProductStack)._getParentStack();
+      this.assetBucket = new ProductStackAssetBucket(parentStack, `ProductStackAssetBucket${hashValues(this.stack.stackName)}`);
+    }
```

Comment:
Do you mean we create a ProductStackAssetBucket regardless of if the user is using an asset or not.
I don't that is a good idea since we would be creating resources that will not be needed (S3-deployment also deploys a lambda). Furthermore, a user could be unintentionally create a bunch of S3 Buckets (since it is one per product-stack) and fill up their S3 Bucket limit capacity.